pkg/k8s/client.go

```diff
@@ -40,13 +40,45 @@ import (
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
 var (
 	// ErrNilNode is returned when the Kubernetes API server has returned a nil node
 	ErrNilNode = goerrors.New("API server returned nil node")
+	client kubernetes.Interface
+	clusterName = "cilium-default"
 )
 
+// GetClusterName returns the cluster name which client is connected to.
+func GetClusterName() string {
+	return clusterName
+}
+
+func setClusterName() {
```

Comment:
This is not the cluster name. The cluster name is defined in `kube-controller-manager` and it's not possible to get it via `kube-apiserver`.

The configuration context is per client, not per cluster. This means different clients can have different context names for the same cluster.

If you really need the cluster name I suggest you add a flag into the cilium CLI and change the cilium daemonset file so that users can specify it using the configmap.

Reply:
I will need a cluster name anyway for the inter cluster routing. Let's add `--cluster-name` and default to `"default"`

Reply:
Ah, darn, OK. I'll fix this up. Thanks for catching that!

Reply:
Done.